plugins/events-backend-module-sqs/config.d.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  events?: {
+    modules?: {
+      /**
+       * events-backend-module-sqs plugin configuration.
+       */
+      awsSqs?: {
+        /**
+         * Contains a record per topic for which an AWS SQS queue
+         * should be used as source of events.
+         */
+        topics: Record<
+          string,
+          {
+            /**
+             * Queue-related configuration.
+             */
+            queue: {
+              /**
+               * (Required) The region of the AWS SQS queue.
+               */
+              region: string;
+              /**
+               * (Required) The absolute URL for the AWS SQS queue to be used.
+               */
+              url: string;
+              /**
+               * (Optional) Visibility timeout for messages in flight.
+               */
+              visibilityTimeoutSeconds: number;
+              /**
+               * (Optional) Wait time when polling for available messages.
+               * Default: 20.
+               */
+              waitTimeSeconds: number;
+            };
+            /**
+             * (Optional) Timeout for the task execution which includes polling for messages
+             * and publishing the events to the event bus.
+             * Must be greater than `queue.waitTimeSeconds`.
+             */
+            timeoutSeconds: number;
```

Comment:
I've seen some other places starting to import HumanDuration and using that in config ... I should move that to `@backstage/types` some day :)

Reply:
with "that" you refer to `HumanDuration`? Were you thoughts around using it here like the following?

```ts
timeout: HumanDuration;
```

For `visibilityTimeoutSeconds` and `waitTimeSeconds` it seems to be reasonable to keep them as they are, esp. the latter.

Reply:
I guess you refer to `TaskScheduleDefinitionConfig` which uses `HumanDuration` and which I added in PRs like PR #14040 etc.

The module already depends on `backend-tasks`, so we could use `HumanDuration` here, too.

At these PRs, we use `readTaskScheduleDefinitionFromConfig` to read the value from the `Config`.
Internally, it uses `readDuration` (supports ISO duration strings) and there `convertToHumanDuration` which are not (yet) exported though.

Re `visibilityTimeoutSeconds` and `waitTimeSeconds`:
we could use something like `Duration.fromObject(value).as('seconds')` then to prevent fractions.